src/Features/Core/Shared/Extensions/DiagnosticAnalyzerExtensions.cs

```diff
@@ -27,7 +28,7 @@ public static DiagnosticAnalyzerCategory GetDiagnosticAnalyzerCategory(this Diag
                 // to be able to operate on a limited span of the document. In practical terms, no analyzer
                 // can have both SemanticDocumentAnalysis and SemanticSpanAnalysis as categories.
                 bool cantSupportSemanticSpanAnalysis = false;
-                var analyzerActions = driver.GetAnalyzerActionsAsync(analyzer).WaitAndGetResult(driver.CancellationToken);
+                var analyzerActions = await driver.GetAnalyzerActionsAsync(analyzer).ConfigureAwait(false);
```

Comment:
calling either Wait or Result on a task in background thread will make that thread wasted until the given task returns.

depends on how heavy the task is, it could sometimes block thread quite long time. so, we try to make sure anything that runs in background never calls Wait/Result.

the worst thing that can happen (which happened before) is that multiple caller wait on heavy tasks on multiple threads causing thread poll to create +1000 threads (which gets worse as it goes since more threads mean slower progress in this situation).

once lesson learned, whenever we find mis-usage like this, we fix it. even though it can propagate quite big.